dandi/upload.py

```diff
@@ -383,3 +398,59 @@ def check_replace_asset(
 
 def skip_file(msg: Any) -> Dict[str, str]:
     return {"status": "skipped", "message": str(msg)}
+
+
+def _bids_discover_and_validate(
+    dandiset_path: str,
+    paths: Optional[List[Union[str, Path]]] = None,
+    validation: Optional[str] = "require",
+) -> List[Path]:
+    """Temporary implementation for discovery and validation of BIDS datasets
+
+    References:
+    - unification of validation records: https://github.com/dandi/dandi-cli/issues/943
+    - validation "design doc": https://github.com/dandi/dandi-cli/pull/663
+    """
+    from .utils import find_files
+    from .validate import validate_bids
+
+    if paths:
+        bids_lookup_paths = set(paths)
+    else:
+        bids_lookup_paths = None
+    bids_descriptions = map(
+        Path, find_files(r"(^|[/\x5C])dataset_description\.json$", dandiset_path)
+    )
+    bids_datasets = [p.parent for p in bids_descriptions]
+    if bids_datasets:
+        lgr.debug(
+            "Detected %s under following paths: %s",
+            pluralize(len(bids_datasets), "BIDS dataset"),
+            ", ".join(str(i) for i in bids_datasets),
+        )
+
+    if validation != "skip":
+        if bids_lookup_paths:
+            bids_datasets_to_validate = list()
+            for p in bids_lookup_paths:
+                if isinstance(p, str):
+                    p = Path(p)
+                for bd in bids_datasets:
+                    try:
+                        p.relative_to(bd)
```

Comment:
This adds `bd` to `bids_datasets_to_validate` if `p` is a child path of `bd`.  Don't we also want to add if `bd` is a child path of `p`?

Reply:
you're right.